docs/how-chugsplash-works.md

```diff
@@ -0,0 +1,252 @@
+* Guarantee that your deployments and upgrades are fully **deterministic**, meaning they can't be halted for any reason.
+* Guarantee that your deployments and upgrades are **atomic**, meaning they're finalized in a single transaction.
+* Approve upgrades of *any* size with a single tiny transaction from a multisig or governance.
+* Keep track of contract dependencies using simple template syntax (`{{ MyContract }}`).
+* Import your own existing proxies into ChugSplash, including Transparent and UUPS proxies.
+
+The shortcomings of today's deployment tooling has led Kelvin and the other engineers at Optimism to spend months developing internal tooling to ensure that their deployments and upgrades are secure. The alternative is to use open source tools that can be unreliable in a production setting. As a result, Optimism upgrades their contracts once every several months, when they'd rather be performing smaller upgrades once a week.
+
+ChugSplash's goal is to enable teams like Optimism to ship smart contracts quickly and iteratively. We think ChugSplash is the future of safe, easy, and predictable smart contract deployments and upgrades.
+
+The ChugSplash Protocol is fully open source and MIT licensed. You can check out the repo [here](https://github.com/chugsplash/chugsplash).
+
+The ChugSplash Managed Service is built on top of the ChugSplash Protocol and includes features such as:
+* Automatic Etherscan verification and ABI publication to npm
+* View proposed upgrades as a diff against the existing system in CI
+* Perform live tests against a proposed upgrade on a forked chain
+* Perform rollbacks to previous versions of your upgraded contracts with the click of a button
+* TODO: anything else?
+
+The managed service is currently in a closed beta. If you'd like to join Optimism and other top protocols in our beta, you can apply [here](https://o7n5gakt66b.typeform.com/to/UoSoli9r). We're accepting only a handful of beta users to ensure that each of them has a fantastic experience.
+
+## How ChugSplash Works
+
+Note: This is about to get technical. You should be proficient with Solidity and have an understanding of proxies and Merkle trees.
+
+### The Brief Version
+
+ChugSplash deployments and upgrades begin when the user defines a [ChugSplash file](https://github.com/chugsplash/chugsplash/blob/develop/docs/chugsplash-file.md). The state variables in the ChugSplash file are converted into 32-byte storage slot key/value pairs using the contract's storage layout. These storage slot key/value pairs are set via `SSTORE` in your proxy. This approach allows us to guarantee that the deployment or upgrade is deterministic, meaning it can't halt for any reason. Since ChugSplash relies on `SSTORE`, all contracts deployed by ChugSplash are proxied. By default, these proxies are upgradeable, but they can always be made non-upgradeable by setting the `admin` to `address(0)`.
```

Comment:
IMO this is confusing because it focuses on only one specific part of the system and ignores everything else. I think we should either rewrite this to cover more of the system at a higher level or remove it entirely. FWIW, I don't think the intended audience is here for the short version.